cpp/src/arrow/compute/exec/plan_test.cc

```diff
@@ -933,6 +933,38 @@ TEST(ExecPlanExecution, SourceGroupedSum) {
   }
 }
 
+TEST(ExecPlanExecution, SourceMinMaxScalar) {
```

Comment:
Is there a task filed for testing all aggregates in this fashion?

Reply:
Preferably we'd update the existing aggregate tests to run all aggregates "both ways" instead of manually duplicating tests like this on a case-by-case basis

Reply:
```suggestion
TEST(ExecPlanExecution, SourceMinMaxScalar) {
  // Regression test for ARROW-16904
```

Reply:
I haven't created such a task, but I was planning on following this up with that, then I can also reference this issue + PR